hibernate-reactive-core/src/test/java/org/hibernate/reactive/MutinySessionTest.java

```diff
@@ -484,6 +484,18 @@ public void testMetamodel(TestContext context) {
 		context.assertEquals( "GuineaPig", pig.getName() );
 	}
 
+	@Test
+	public void reactiveThreadBoundTransaction(TestContext context) {
+		test( context, getMutinySessionFactory().withTransaction(
+				(session, transaction) -> session.createQuery("from GuineaPig").getResultList().invoke( list -> {
+					context.assertNotNull( session.currentTransaction() );
```

Comment:
Shouldn't be `assertTrue( session.currentTransaction() == transaction)`?

Reply:
Well, I don't think we care if the actual object is identically equal, only that it represents the same _state_.

Reply:
Ah right, makes sense. Thanks